src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -158,11 +158,12 @@ impl LogicalAgg {
             ));
             Ok(global_agg.into())
         } else {
-            let exchange = RequiredDist::shard_by_key(input_col_num, self.group_key())
+            let group_key = (0..self.group_key().len()).collect_vec();
+
+            let exchange = RequiredDist::shard_by_key(input_col_num, &group_key)
                 .enforce_if_not_satisfies(local_agg.into(), &Order::any())?;
             // Local phase should have reordered the group keys into their required order.
             // we can just follow it.
```

Comment:
```suggestion
```

Reply:
I guess this is correct. In some other places, we use `ColIndexMapping` and `mapping.rewrite_dist_key(&in_dist_key)` to achieve this column mapping.

Reply:
I would suggest:

```rust
    let global_group_key = local_agg.i2o_col_mapping().rewrite_dist_key(self.group_key()).unwrap();
```

This seems to be logically precise.🧐